### PR TITLE
TrustedForm URL

### PR DESCRIPTION
**Pull request by alexkwolfe**
## Description of the change

This adds a new data type called `trustedform_url`. The `trustedform_cert_url` should be converted to this type. In production, one or two customers are using the `host` and `path` components, so I've left those. All the other components of the `url` type are not necessary.

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [X] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Configuration change
- [ ] Technical Debt
- [ ] Documentation

## Related tickets

[Shortcut #37783](https://app.shortcut.com/active-prospect/story/37783/trustedform-cert-url-type)

### Development and Testing

- [ ]  Lint rules pass locally.
- [ ]  The code changed/added as part of this pull request has been covered with tests, or this PR does not alter production code.
- [ ]  All tests related to the changed code pass in development, or tests are not applicable.

### Code Review

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached.
- [ ]  At least two engineers have been added as "Reviewers" on the pull request.
- [ ]  Changes have been reviewed by at least two other engineers who did not write the code.
- [ ]  This branch has been rebased off master to be current.

### Tracking 
- [ ]  Issue from Shortcut has a link to this pull request.
- [ ]  This PR has a link to the issue in Shortcut.

### QA
- [ ]  This branch has been deployed to staging and tested.

### CODE REVIEW

File: test/trustedform-url.test.js [hunk 1]
```diff
@@ -0,0 +1,188 @@
+const {assert} = require('chai');
+const moment = require('moment');
+
+// these environment variables must be set before requiring the type
```
Comment (cgrayson) [inline test/trustedform-url.test.js:4]: Hmm. Does this mean that LC won't launch without having values for these?

**Review by cgrayson — CHANGES_REQUESTED** [review]
Nice. A couple minor things inline, plus you can delete `package-lock.json`. It's ignored for anything but the top-level package being installed, and so is omitted from dependent libraries like this one. (Also, icymi: we can't "approve" PRs & let the author fix & merge anymore; after you make changes, this review will be stale and you'll have to re-request another one.)

File: test/trustedform-url.test.js (hunk 1 above)
Comment (alexkwolfe) [reply test/trustedform-url.test.js:4]: LC will launch, but the type won't parse TF URLs because the private keys won't be set.

**Review by cgrayson — APPROVED** [review]
LGTM

**Pull request merged**